keps/sig-node/4540-strict-cpu-reservation/README.md

```diff
@@ -0,0 +1,509 @@
+                if podRequest.ExclusiveMilliCPU > 0 && podRequest.ExclusiveMilliCPU > (nodeInfo.Allocatable.ExclusiveMilliCPU-nodeInfo.Requested.ExclusiveMilliCPU) {
+                        insufficientResources = append(insufficientResources, InsufficientResource{
+                                ResourceName: v1.ResourceExclusiveCPU,
+                                Reason:       "Insufficient exclusive cpu",
+                                Requested:    podRequest.ExclusiveMilliCPU,
+                                Used:         nodeInfo.Requested.ExclusiveMilliCPU,
+                                Capacity:     nodeInfo.Allocatable.ExclusiveMilliCPU,
+                        })
+                }
+        }
+```
+
+#### Archived Risk Mitigation (Option 2)
+
+The problem with `MinSharedCPUs` is that it creates another complication like memory and hugpages, new resources vs overlapping resources, exclusive-cpus is a subset of cpu.
+
+Currently the noderesources scheduler plugin does not filter out the best-effort pods in the case there's no available CPU.
+
+Another option is to force the cpu requests for best effort pods to 1 MilliCPU in kubelet for the purpose of resource availability checks (or, equivalently, check there's at least 1 MilliCPU allocatable). This option is meant to be simpler than option-1, but it can create runaway pods similar to that in https://github.com/kubernetes/kubernetes/issues/84869.
+
+
+### Test Plan
+
+[X] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this enhancement.
+
+##### Prerequisite testing updates
+
+<!--
+Based on reviewers feedback describe what additional tests need to be added prior
+implementing this enhancement to ensure the enhancements have also solid foundations.
+-->
+
+##### Unit tests
+
+- `k8s.io/kubernetes/pkg/kubelet/cm/cpumanager/policy_static.go`: `03-18-2024` - `91.1`
+
+##### Integration tests
+
+- These cases will be added in the existing integration tests:
+  - Feature gate enable/disable tests
+  - `strict-cpu-reservation` policy option works as expected.
+  - `strict-cpu-reservation` policy option works with existing policy options.
+
+##### e2e tests
+
+- These cases will be added in the existing e2e tests:
+  - Feature gate enable/disable tests
+  - `strict-cpu-reservation` policy option works as expected.
+  - `strict-cpu-reservation` policy option works with existing policy options.
+
+### Graduation Criteria
+
+#### Alpha
+
+- [X] Implement the new policy option.
+- [X] Ensure proper unit tests are in place.
+- [X] Ensure proper e2e node tests are in place.
+
+#### Beta
+
+- [X] Gather feedback from consumers of the new policy option.
+- [X] Verify no major bugs reported in the previous cycle.
+
+#### GA
+
+- [X] Allow time for feedback (1 year).
+- [X] Make sure all risks have been addressed.
```

Comment:
Nit the checkboxes above are confusing, I'd remove them entirely. Or at least check the ones which apply to this document.

Reply:
I removed the checks for Beta and GA stages.